src/utils/use-reduced-motion.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState, useContext } from "react"
+import { prefersReducedMotion, shouldReduceMotion } from "../dom/accessibility"
+import { MotionContext } from "../motion/context/MotionContext"
+
+/**
+ * A hook that returns `true` if we should be using reduced motion based on the current device's Reduced Motion setting.
+ *
+ * This can be used to implement changes to your UI based on Reduced Motion. For instance, replacing motion-sickness inducing
+ * `x`/`y` animations with `opacity, disabling the autoplay of background videos, or turning off parallax motion.
+ *
+ * It will actively respond to changes and re-render your components with the latest setting.
+ *
+ * ```jsx
+ * export function MyComponent() {
+ *   const isReducedMotion = useReducedMotion()
+ *
+ *   return <video autoplay={!isReducedMotion} />
+ * }
```

Comment:
I'd suggest going with an example that uses motion here such as replacing the x/y animation with opacity.